### PR TITLE
Use vue for the homepage

### LINKED ISSUE #259: Total pages
There is only arrows for navigation. Please, add navigation using page numbers.
Like in github issues page:
<< 1 2 3 ... 99 >>

So, we can see, how much pages we have

### PR DESCRIPTION
**Pull request by felixoi**
Opened to keep track of the stuff. Not fully ready yet.

Fixes #259

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -190,8 +190,48 @@ def flexmarkDep(module: String) = {
...
+      "babel-loader"                       -> "8.0.6",
+      "@babel/core"                        -> "7.4.5",
+      "terser-webpack-plugin"              -> "1.3.0",
+      "mini-css-extract-plugin"            -> "0.7.0",
+      "optimize-css-assets-webpack-plugin" -> "5.0.1",
+      "sass-loader"                        -> "7.1.0",
+      "postcss-loader"                     -> "3.0.0",
+      "autoprefixer"                       -> "9.5.1",
+      "node-sass"                          -> "4.12.0",
+      "query-string"                       -> "6.5.0",
```
Comment (felixoi) [inline build.sbt]: This shouldnt be a dev dependency.

File: oreClient/src/main/resources/assets/components/ProjectList.vue [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+            <i class="fas fa-spinner fa-spin"></i>
+            <span>Loading projects for you...</span>
+        </div>
+        <div v-show="!loading">
+            <ul class="list-group project-list">
+                <li v-for="project in projects" class="list-group-item project @entry.visibility.cssClass">
+                    <div class="container-fluid">
+                        <div class="row">
+                            <div class="col-xs-12 col-sm-1">
+                                <img class="user-avatar user-avatar-sm" :src="project.icon_url" :alt="project.name" />
```
Comment (felixoi) [inline oreClient/src/main/resources/assets/components/ProjectList.vue]: Note to myself to make this clickable again.

File: build.sbt [hunk 3]
```diff
@@ -190,8 +190,48 @@ def flexmarkDep(module: String) = {
   "com.vladsch.flexmark" % artifactId % flexmarkVersion
 }
 
+lazy val oreClient = project
+  .enablePlugins(ScalaJSBundlerPlugin)
+  .settings(
+    name := "ore-client",
+    commonSettings,
+    useYarn := true,
+    scalacOptions += "-P:scalajs:sjsDefinedByDefault",
```
Comment (Katrix) [inline build.sbt]: Can probably delete this

File: ore/public/javascripts/apiRequests.js [hunk 4]
```diff
@@ -3,31 +3,23 @@
...
 let isLoggedIn = false;
 
+//=====> SETUP
+
+jQuery.ajaxSettings.traditional = true;
+
 //=====> HELPER FUNCTIONS
 
-function apiV2Request(url, method, data, isRetry) {
+function apiV2Request(url, method = "GET", data = {}, isRetry = true) {
```
Comment (Katrix) [inline ore/public/javascripts/apiRequests.js]: From what I remember when I wrote this function, isRetry should only be true when it has already tried the request once, failed, and gotten a new token. As it is now it breaks loading if the user isn't logged in.

File: oreClient/src/main/resources/assets/Home.vue [hunk 5]
```diff
@@ -0,0 +1,205 @@
+<template>
+    <div class="row">
+        <div class="col-md-9">
+            <div class="project-search">
+                <input type="text" class="form-control" v-model="q" :placeholder="'Search in ' + projectCountPlaceholder + ' projects, proudly made by the community...'" />
```
Comment (Katrix) [inline oreClient/src/main/resources/assets/Home.vue]: Do we want to update the project count when a filter is applied too?

File: oreClient/src/main/resources/assets/components/Pagination.vue [hunk 6]
```diff
@@ -0,0 +1,84 @@
+<template>
```
Comment (Katrix) [inline oreClient/src/main/resources/assets/components/Pagination.vue:1]: Can we make a prettier pagination for the home page? It just doesn't feel like it fits at all. The colors are all wrong, and it feels so small. At least make it a bit bigger/longer.

Comment (Katrix) [inline oreClient/src/main/resources/assets/components/Pagination.vue:1]: This breaks when on the second page and searching for something which only returns one page.

File: ore/public/javascripts/apiRequests.js (hunk 4 above)
Comment (Katrix) [inline ore/public/javascripts/apiRequests.js]: From what I remember when I wrote this function, isRetry should only be true when it has already tried the request once, failed, and gotten a new token. As it is now it breaks loading if the user isn't logged in.

File: oreClient/src/main/resources/assets/components/Pagination.vue (hunk 6 above)
Comment (felixoi) [reply oreClient/src/main/resources/assets/components/Pagination.vue:1]: Yeah just copied the design for the version page

File: oreClient/src/main/resources/assets/Home.vue (hunk 5 above)
Comment (felixoi) [reply oreClient/src/main/resources/assets/Home.vue]: Isn't this updating? It's bound to the pagination.count of the last request. So if a filter is applied pagination.count should have a different value or not?

File: ore/public/javascripts/apiRequests.js (hunk 4 above)
Comment (felixoi) [reply ore/public/javascripts/apiRequests.js]: Yeah indeed. Default should be false.

Comment (Katrix) [reply ore/public/javascripts/apiRequests.js]: Actually, when I think about it a bit more, the whole idea of "Fire off a request and see if it succeeds" is stupid when the authentication request includes the expiration date. Maybe rewrite stuff to prefer to use that?

File: oreClient/src/main/resources/assets/Home.vue (hunk 5 above)
Comment (Katrix) [reply oreClient/src/main/resources/assets/Home.vue]: What I more meant is if this is the behavior we want, or if we want the count to  only change when no filters are used.

Comment (felixoi) [reply oreClient/src/main/resources/assets/Home.vue]: I'd say we should update it. Will push a commit later I've locally already which changes the message if filters are active but still updates the count.

Comment (felixoi) [reply oreClient/src/main/resources/assets/Home.vue]: Can you recheck if this fits now?

**Review by Katrix — APPROVED** [review]

**Pull request merged**